src/mo_interpreter/interpret.ml

```diff
@@ -885,16 +894,31 @@ and match_shared_pat env shared_pat c =
 
 (* Objects *)
 
-and interpret_obj env obj_sort dec_fields (k : V.value V.cont) =
+and interpret_obj env obj_sort self_id dec_fields (k : V.value V.cont) =
   match obj_sort with
   | T.Actor ->
      let self = V.fresh_id() in
+     let self' = V.Blob self in
+     Option.iter (fun id -> define_id env id self') self_id;
      let ve_ex, ve_in = declare_dec_fields dec_fields V.Env.empty V.Env.empty in
-     let env' = adjoin_vals { env with self = self } ve_in in
+     let env' = adjoin_vals { env with self } ve_in in
```

Comment:
I think if you declare_id self (if necessary before adding ve_in to env you might be ok.

Reply:
Yeah!